polar-core/src/vm.rs

```diff
@@ -1926,6 +1931,74 @@ impl PolarVirtualMachine {
         Ok(QueryEvent::None)
     }
 
+    /// Check for partially bound or unbound variable arguments to an external call.
+    /// If any arguments are partially bound or unbound, return an error.
+    /// The only exception is the method `OsoRoles.role_allows()`, which expects a partially bound resource argument
+    fn check_partial_args(
+        &mut self,
+        object: &Term,
+        field: &Term,
+        value: &Term,
+    ) -> PolarResult<bool> {
+        // If the lookup is a `Value::Call`, then we need to check for partial args
+        let mut has_partial_arg = false;
+        if let Value::Call(Call { name, args, kwargs }) = field.value() {
+            // get all arg values (args + kwargs)
+            let mut arg_values = args.clone();
```

Comment:
Do we need all these clones?

Reply:
I think so? It took me a while to make this happy, but I'd be glad to pair on cutting down the clones